iso15118/secc/states/iso15118_2_states.py

```diff
@@ -864,24 +871,66 @@ async def process_message(
             # TODO Either an MO Root certificate or a V2G Root certificate
             #      could be used to verify, need to be flexible with regards
             #      to the PKI that is used.
-            verify_certs(leaf_cert, sub_ca_certs, load_cert(CertPath.MO_ROOT_DER))
-
-            # TODO Check if EMAID has correct syntax
-
+            # TODO GitHub#94: If root_cert is not present, we should
+            #      fall back to sending the leaf and sub-CA certificates,
+            #      allowing the CSMS to attempt to retrieve the root certificate
+            #      and construct the OCSP data itself.
+            root_cert_path = self._mobility_operator_root_cert_path()
+            root_cert = load_cert(root_cert_path)
+            verify_certs(leaf_cert, sub_ca_certs, root_cert)
+
+            # Note that the eMAID format (14 or 15 characters) will be validated
+            # by the definition of the eMAID type in
+            # shared/messages/iso15118_2/datatypes.py
+            self.comm_session.emaid = payment_details_req.emaid
             self.comm_session.contract_cert_chain = payment_details_req.cert_chain
 
-            payment_details_res = PaymentDetailsRes(
-                response_code=ResponseCode.OK,
-                gen_challenge=get_random_bytes(16),
-                evse_timestamp=time.time(),
+            hash_data = get_certificate_hash_data(
+                self.comm_session.contract_cert_chain, root_cert
+            )
+            pem_certificate_chain = build_pem_certificate_chain(
+                self.comm_session.contract_cert_chain, root_cert
             )
 
-            self.create_next_message(
-                Authorization,
-                payment_details_res,
-                Timeouts.V2G_SECC_SEQUENCE_TIMEOUT,
-                Namespace.ISO_V2_MSG_DEF,
+            authorization_result = (
+                await self.comm_session.evse_controller.is_authorized(
+                    id_token=payment_details_req.emaid,
+                    id_token_type=AuthorizationTokenType.EMAID,
+                    certificate_chain=pem_certificate_chain,
+                    hash_data=hash_data,
+                )
             )
+
+            if authorization_result == AuthorizationStatus.ACCEPTED:
+
+                payment_details_res = PaymentDetailsRes(
+                    response_code=ResponseCode.OK,
+                    gen_challenge=get_random_bytes(16),
+                    evse_timestamp=time.time(),
+                )
+
+                # TODO: Should the next message really be Authorization?  If so why?
```

Comment:
If the Auth status is Accepted or Ongoing, yes, the next step shall be the Authorization class, as that is the message the EV will send after reception of the PaymentDetailsRes